### PR TITLE
[feature] Scrolling list directive

### PR DESCRIPTION
**Pull request by alex-deaconu**
Split out code from the CH pertaining to Lists and Scroll events
Created `scrolling-list` directive which wraps the Scroll event along w/ verification for hitting the bottom of the page
`BaseList` functionality & tests, removed un-used functions (need to confirm)

### CODE REVIEW

**Comment by alex-deaconu** [comment]
This is my version of a `scrolling-list` directive. It basically grabs the existing functionality from the CH, wraps it in a nicer directive, fixes a few issues among which https://github.com/Rise-Vision/common-header/issues/492 and adds unit/e2e test coverage.

This is a first iteration, going forward we can make it fancier if we want.

You can see the directives in action here: https://github.com/Rise-Vision/common-header/issues/492.

@olegrise @rodrigopavezi FYI.

@tejohnso please review. Thanks!

File: test/unit/scrolling-list/base-list.tests.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    };
+    
+    inject(function($injector){
+      var BaseList = $injector.get("BaseList");
+      listObject = new BaseList();
+    });
+  });
+  
+  it("should exist",function(){
+    expect(listObject).to.be.truely;
```
Comment (tejohnso) [inline test/unit/scrolling-list/base-list.tests.js]: I thought that the tests weren't running when I saw "truely" but it turns out it doesn't matter what is used there.  I changed it to "exists" and it worked, then changed it to "to.not.exist" and it failed.  Then I changed it to "to.be.spock" and it worked :)

**Comment by tejohnso** [comment]
:+1:

File: test/unit/scrolling-list/base-list.tests.js (hunk 1 above)
Comment (alex-deaconu) [inline test/unit/scrolling-list/base-list.tests.js]: @tejohnso I don't even know where I got `truely` from. Changed to `ok` and verified that it does indeed fail if I set it reversed. Thanks for catching that!

**Pull request merged**